### PR TITLE
keystore: sign & verify with prefix and date

### LINKED ISSUE #191: Add prefix & date on msg before signing
To avoid attacks of the type:
- userA sends a msg to userB requesting a signature
- userB signs msg thinking about X
- userA receives msg signed, and that msg was a root-update packet, so userA has that non legit packet signed by userB

To avoid this and other kind of attacks, we'll use a prefix in msgs before hashing it and signing.
For each type of msg to be signed there will be a specific prefix.

Edit:
msg to be signed will be:
```
prefix | date | original_msg
```

### PR DESCRIPTION
**Pull request by arnaucube**
resolves #191

### CODE REVIEW

File: keystore/keystore.go [hunk 1]
```diff
@@ -45,6 +48,14 @@ const (
 	scryptDKLen = 32
 )
 
+// prefixes for msg to be signed
+type PrefixType []byte
+
+var (
+	// PrefixRootUpdate is for signatures related to update the root of an identity
+	PrefixRootUpdate = []byte("rootupdate")
```
Comment (ed255) [inline keystore/keystore.go]: Maybe "rootminorupdate"?

File: keystore/keystore.go [hunk 2]
```diff
@@ -363,9 +384,18 @@ func VerifySignatureElem(pkComp *babyjub.PublicKeyComp, msg mimc7.RElem, sigComp
 	return pk.VerifyMimc7(msg, sig), nil
 }
 
-// VerifySignatureElem verifies that the signature sigComp of the mimc7 hash of
+// VerifySignature verifies that the signature sigComp of the mimc7 hash of
+// the [prefix | date | msg] byte slice was signed with the public key pkComp.
+func VerifySignature(pkComp *babyjub.PublicKeyComp, sigComp *babyjub.SignatureComp, prefix PrefixType, date int64, rawMsg []byte) (bool, error) {
+	msg := append(prefix, utils.Uint64ToEthBytes(uint64(date))...)
+	msg = append(msg, rawMsg...)
+	h := mimc7HashBytes(msg)
```
Comment (ed255) [inline keystore/keystore.go]: why not use VerifySignatureRaw here?

File: keystore/keystore.go (hunk 1 above)
Comment (arnaucube) [reply keystore/keystore.go]: right! done as `minorupdate`, as the concept only exists for roots (identity state)

File: keystore/keystore.go (hunk 2 above)
Comment (arnaucube) [reply keystore/keystore.go]: right! done

**Pull request merged**